File: src/data_structure/trees/level_traversal.cpp

/*
struct node
{
    int data;
    node* left;
    node* right;
}*/

void LevelOrder(node * root)
{
    int count = 0;
    struct node *nodes[10000];
    nodes[count++] = root;
    while(count > 0) {
        node *n = nodes[--count];
        cout << n->data << " ";
        if (n->left != NULL) {
            nodes[count++] = n->left;
        }
        if (n->right != NULL) {
            nodes[count++] = n->right;
        }
    }
}

File: src/data_structure/trees/level_traversal.java
void LevelOrder(Node root) {
    Queue<Node> q = new LinkedList<Node>();
    q.add(root);
    while (!q.isEmpty()) {
        Node n = q.remove();
        System.out.print(n.data + " ");
        if (n.left != null) {
            q.add(n.left);
        }
        if (n.right != null) {
            q.add(n.right);
        }
    }
}
